### DISCUSSION TITLE
Standardization of rule name

### DISCUSSION THREAD

**Discussion by picklebento** [detekt/detekt#3448] [Ideas]
There are 10 rules with `Unnecessary` and 5 rules with `Redundant` in their names, we should standardize them so that those code simplification rules are in vicinity in the YAML config.

Note: I discovered this when I thought `RedundantHigherOrderMapUsage` is #3341.

**Comment by picklebento** [comment]
There are also 2 rules with `Useless`

**Comment by BraisGabin** [comment]
I completely agree. We need to review all our rules and unify the naming. And create a guide so the new rules can follow some guidelines. And something similar with the configurations.

We should do this work before 2.0 so we know the renaming that we want and the apply it once we are ready to release it.

#2586 is related with this

**Reply by cortinico** [reply]
> We should do this work before 2.0

So we will basically use aliases and do the full rename in 2.x?

**Reply by BraisGabin** [reply]
No, with "this work" I mean to find all the inconsistent things that we don't like and write and small guide about how name rules and flags. For example: don't use negative. We can look at more mature projects and see if they have something similar.

**Comment by schalkms** [comment]
I'm not a fan of doing this for several reasons as I also mentioned in the linked issue. Didn't we already discuss this through?

> As a user of detekt, I'm tired of constantly changing my existing config just for the sake of change. It costs time. I'm not sure regarding the user benefit of having a uniform naming.

**Reply by cortinico** [reply]
Valid point @schalkms 

> Didn't we already discuss this through?

Can you link the Issue/PR?

**Reply by picklebento** [reply]
https://github.com/detekt/detekt/issues/2586#issuecomment-610272791

**Reply by BraisGabin** [reply]
And I agree with it. But if we are going to move to a kotlin dsl in 2.0 (that's not for sure yet but I'd like to) the change will be so big that this rename would be trivial. I agree that between minor updates we should not change anything. But if we talk about a major that will already disrupt that part we can use it.
